### DISCUSSION TITLE
How can i display table in my custom view?

### DISCUSSION THREAD

**Discussion by Samoylenko666** [filamentphp/filament#5553] [Help]
According to the documentation I created the following custom view
```
use App\Filament\Resources\ProductResource;
use App\Models\Product;
use Filament\Resources\Pages\ViewRecord;
use Filament\Tables\Concerns\InteractsWithTable;
use Filament\Tables\Contracts\HasTable;
use Illuminate\Contracts\View\View;
use Illuminate\Database\Eloquent\Builder;

class ViewProduct extends ViewRecord implements HasTable
{
    use InteractsWithTable;

    protected static string $resource = ProductResource::class;

    protected static string $view = 'filament.pages.product.view-record';

    public function render(): View
    {
        return view('filament.pages.product.view-record');
    }

    protected function getTableColumns(): array
    {
        return ['name'];
    }

    protected function getTableQuery(): Builder
    {
         return Product::query()->where('id',1);
    }
}
```

now in my view i use `<div>
        {{ $this->table }}
    </div>`

But this get me error 
`local.ERROR: Typed property App\Filament\Resources\ProductResource\Pages\ViewProduct::$table must not be accessed before initialization {"view":{"view":"/var/www/resources/views/filament/pages/product/view-record.blade.php","data":{"errors":"<pre class=sf-dump id=sf-dump-277303290 data-indent-pad=\"  \"><span class=sf-dump-note>Illuminate\\Support\\ViewErrorBag</span> {<a class=sf-dump-ref>#1433</a><samp data-depth=1 class=sf-dump-expanded>
  #<span class=sf-dump-protected title=\"Protected property\">bags</span>: []
</samp>}.....`

How can I fix this?
Should I define the $table variable in the mount method?

**Comment by danharrin** [comment]
Very weird, it looks correct 🤔 Create a reproduction repo if you want us to help

**Comment by bubaololo** [comment]
Got the same error

**Comment by Sweeeeep** [comment]
Got the same error as well

**Comment by danharrin** [comment]
@Samoylenko666 @bubaololo @Sweeeeep I would not advise using the table builder directly on the page. Create a separate Livewire component which uses it, and embed it in the page view.

**Comment by bubaololo** [comment]
In my another attempt to create custom page with table i create the standalone page and resouce page, then copy the declaration from resource page to standalone and finally it works.

https://github.com/bubaololo/ortomaster/blob/master/app/Filament/Pages/TodayAppointments.php

**Reply by danharrin** [reply]
I really wouldn't recommend this, a separate Livewire component is the way to go

**Reply by adamlast** [reply]
Can you please link to an example or tutorial on how to do that

**Reply by howdu** [reply]
@adamlast It's all in the documentation but this is a clean tutorial 
[https://devdojo.com/codercodyjenson/using-the-filament-table-builder](https://devdojo.com/codercodyjenson/using-the-filament-table-builder)

After you've created the table component you can display it within Filament with`@livewire('custom-table')`

**Reply by adamlast** [reply]
@howdu Thanks.  I can work out from the codebase that @livewire('custom-table') goes in a blade file but I'm really struggling with all the filament functionality for blade files.  I just really need a cheat sheet for how the Filament blade components are meant to work.

**Reply by danharrin** [reply]
What would you want the cheatsheet to contain?

**Reply by adamlast** [reply]
@danharrin Good question. Not 100% sure but I'd like to know what type of things can go into blade files and how they interact with the Filament PHP components.  I'm building up a table widget with some customised buttons next to it. It's starting to look ok and work correctly but I'm doing lots of diving through the code base and generated files to figure out what to do.

**Reply by danharrin** [reply]
I'm sorry, but I guess I don't know what is missing from the docs, because everything to get started should be in there

**Comment by izdrail** [comment]
yes all good , but you still need a builder :|
